core/kazoo_media/src/kz_media_map.erl

```diff
@@ -123,13 +123,20 @@ handle_media_doc_change(JObj, _Change) ->
     {'ok', Doc} = kz_datamgr:open_doc(Db, kz_json:get_value(<<"ID">>, JObj)),
     gen_listener:cast(?MODULE, {'add_mapping', Db, Doc}).
 
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Parse the kz_media_map prompt ID from media doc ID.
+%% Keeping old format separator "%2F" for backwards compatibility,
+%% i.e. deleting prompts carried forward from Kazoo < 4.0
+%% @end
+%%--------------------------------------------------------------------
 -spec extract_prompt_id(kz_term:ne_binary()) -> kz_term:ne_binary().
-extract_prompt_id(<<_Lang:5/binary, "%2F", PromptId/binary>>) ->
-    PromptId;
-extract_prompt_id(<<_Lang:2/binary, "%2F", PromptId/binary>>) ->
-    PromptId;
-extract_prompt_id(MediaId) ->
-    MediaId.
+extract_prompt_id(<<_Lang:5/binary, "/", PromptId/binary>>) -> PromptId;
+extract_prompt_id(<<_Lang:2/binary, "/", PromptId/binary>>) -> PromptId;
+extract_prompt_id(<<_Lang:5/binary, "%2F", PromptId/binary>>) -> PromptId;
+extract_prompt_id(<<_Lang:2/binary, "%2F", PromptId/binary>>) -> PromptId;
+extract_prompt_id(MediaId) -> MediaId.
```

Comment:
@danielfinke why are adding this ? risk of going back to double encoding. we need example examples before accepting this. i don't like the risk.

Reply:
@lazedo this function is only called by the ?DOC_DELETED clause of handle_media_doc_change/2. Will only contain an ID of whatever the deleted doc was. I'm keeping the old clauses ("%2F") for accounts that were brought forward from Kazoo 3.22 and below. The stuff I added ("/") fixes the function for new prompt overrides, since those docs' ids contain "/". kz_media_map shouldn't be getting a prompt_id that contains the lang, because that's not how it's stored in the ets table.